pkg/resource/deploy/step.go

```diff
@@ -960,7 +963,29 @@ func (s *RefreshStep) Apply(preview bool) (resource.Status, StepCompleteFunc, er
 			now := time.Now().UTC()
 			s.new.Modified = &now
 		}
+
+		// We compute the diff that a user would see if they immediately ran an `up` on a no-change program
+		// after this refresh.  This will return the opposite of what we want - so we'll invert it below.
+		// We pass:
+		// * newInputs where oldInputs are expected
+		// * newOutputs where oldOutputs are expected
+		// * oldInputs where newInputs are expected
+		diff, err := diffResource(s.new.URN, s.new.ID, s.new.Inputs, s.new.Outputs, s.old.Inputs, prov, preview,
```

Comment:
Assuming this is going to call Diff(), I love that because that automatically engages bridge machinery for planning changes suppressing changes and computing detailed diffs, that is going to be very helpful to run that. The downside is that new inputs returned from Read() are not check-clean and Diff expects these to be checked clean inputs so this may uncover some latent issues in bridged providers when rolled out.

Reply:
> new inputs returned from Read() are not check-clean 

I think the implied requriement even before these changes is that they are, as they have always been stored directly into the state file, and only "check-clean" inputs are allowed to be stored in state.  And that's a reasonable requirement, because these are inputs provided directly by a provider, which is capable of ensuring anything it returns is check-clean. Check is only needed to take an arbitrary bag of inputs from a users and turn it into something the provider knows is valid.